tests/test_loss.py

```diff
@@ -27,3 +27,35 @@ def test_pseudo_huber_loss():
 
 def test_flat_bottom_loss():
     _assert_basic_loss_properties(flat_bottom_loss)
+
+
+def _scalar_truncated_residual(prediction, label, reliable_interval):
+    lower, upper = reliable_interval
+    assert lower < upper
+
+    if (label >= lower) and (label <= upper):
+        residual = prediction - label
+    elif label < lower:
+        residual = max(0, prediction - lower)
+    elif label > upper:
+        residual = min(0, prediction - upper)
+    else:
+        raise (RuntimeError('unsatisfiable reliable_range'))
+
+    return residual
+
+
+def test_compute_residuals():
+    predictions = np.random.rand(10000) * 4 - 2
+    labels = np.random.rand(10000) * 4 - 2
+
+    reliable_interval = (-1, +1)
+
+    ref = np.array([
+        _scalar_truncated_residual(p, l, reliable_interval)
+        for (p, l) in zip(predictions, labels)
+    ])
+
+    test = truncated_residuals(predictions, labels, reliable_interval)
```

Comment:
possible to also add a test for [-np.inf, +np.inf] interval?

Reply:
Addressed in https://github.com/proteneer/timemachine/pull/480/commits/f12351e2aa7e50010073c5819964427742d91396